dashboard/src/components/ErrorAlert/ErrorSelector.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from "react";
+import { definedNamespaces } from "../../shared/Namespace";
+import {
+  AppConflict,
+  ForbiddenError,
+  IRBACRole,
+  NotFoundError,
+  UnprocessableEntity,
+} from "../../shared/types";
+import { PermissionsErrorAlert, UnexpectedErrorAlert } from "./";
+import { namespaceText } from "./helpers";
+
+interface IErrorSelectorProps {
+  error: Error;
+  // Action related to the error (e.g. create, delete)
+  action?: string;
+  // Default permissions that the component should have
+  // returned in case it's a ForbiddenError and the message
+  // doesn't contain specific RBAC roles
+  defaultRequiredRBACRoles?: {
+    [action: string]: IRBACRole[];
+  };
+  // Namespace in case the error is namespaced
+  namespace?: string;
+  // Resource causing the failure (e.g. "Application my-wordpress")
+  resource?: string;
+}
+
+class ErrorSelector extends React.Component<IErrorSelectorProps> {
```

Comment:
As a personal preference, I'd try to define components as Stateless functional component (SFC) when possible.